### DISCUSSION TITLE
Ship frames being changed?

### DISCUSSION THREAD

**Discussion by SMG001** [bcholmes/StarTrek2d20#104] [General]
I was trying to create a new starship for my campaign, I got surprised by the change in the UI in the site, but nonetheless I continued making the ship, until I got to the export PDF file, I went to check the ship and I discovered that my Gagarin-Class starship suddenly turned into a Galaxy-Class Starship, I'd like to know if anyone else has had this problem

**Comment by bcholmes** [comment]
I don’t have an illustration for a Gagarin class; the Galaxy ends up being the default picture.

Is it just the picture that’s showing a Galaxy or are you seeing that elsewhere?

**Reply by SMG001** [reply]
I think it's just the picture, since I tried making a Constellation and a Constitution Class and they are apparently normal

**Reply by Felderburg** [reply]
> I don’t have an illustration for a Gagarin class; the Galaxy ends up being the default picture.

I swear I remember you posting the process for getting ship illustrations somewhere, but now I can't find it. What is the process?

**Reply by Felderburg** [reply]
> > I don’t have an illustration for a Gagarin class; the Galaxy ends up being the default picture.
> 
> I swear I remember you posting the process for getting ship illustrations somewhere, but now I can't find it. What is the process?

Haha, found it! https://mastodon.social/@bcholmes/108419805135877979 Is there a way to help out with this? You say "My PDF library lets me draw paths using the SVG path notation"... it's not clear to me if you're taking the SVG in something like illustrator, converting it to the html `<path>` that your pdf library can use, or if your pdf library converts it to the `<path>`. Either way, would providing svgs be of help?

(For reference for anyone else: https://www.w3schools.com/graphics/svg_path.asp) and https://github.com/bcholmes/StarTrek2d20/blob/sta-complete/WebTools/src/helpers/spaceframeOutlineHelper.ts)

**Reply by bcholmes** [reply]
The number one thing that I'm struggling to source is 3D models of some of the spaceframes. For example, I haven't been able to put my hands on:

- Warp Delta
- J-Class/Y-Class
- Gagarin Class
- Reliant Class
- Raven type

And most of the Section 31 frames from the Discovery book. Most of the Klingon frames from that book are also things I haven't been able to get my hands on. 

When I say that the library lets me draw path, what I mean is that SVG often uses a [tag called "path"](https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Paths) which includes a series of point-to-point instructions, like so: `d="M 10 10 h 80 v 80 h -80 Z"` This particular path is understood as "move the starting point to 10,10, then move horizontally 80, then vertically 80 and then horizontally -80 and then close the loop by drawing a straight line to the starting position".

I can get the PDF library to process those point-to-point instructions. (If you look at the spaceframeOutlineHelper file, you'll see it starts with a long list of instructions for each of the spaceframes it knows about). Those instructions basically tell the PDF how to move a virtual pen around the page to produce the ship silhouette.

**Reply by bcholmes** [reply]
As for the question, "would providing SVGs be of help?"... maybe? If they're of comparable quality and about the same number of points, they might work. I'd probably have questions about provenance.

**Reply by Felderburg** [reply]
> As for the question, "would providing SVGs be of help?"... maybe? If they're of comparable quality and about the same number of points, they might work. I'd probably have questions about provenance.

I was thinking I would trace things myself... the 3/4 view would be a potential issue. I noticed just now that some code tags I put in the comment above were left out because I didn't notate them as code (I don't think it matters for your reply).
